generators/entity/index.js

```diff
@@ -601,6 +601,13 @@ class EntityGenerator extends BaseBlueprintGenerator {
         this.context.entityContainsCollectionField = this.context.relationships.some(relationship => relationship.relationshipCollection);
       },
 
+      processPrimaryKeyTypesForRelations() {
+        const types = this.context.relationships.map(
+          relationship => relationship.otherEntity.primaryKey && relationship.otherEntity.primaryKey.type
+        );
```

Comment:
```suggestion
        const types = this.context.relationships
            .filter(relationship => relationship.otherEntity.primaryKey)
            .map(relationship => relationship.otherEntity.primaryKey.type);
```
Making sure we are not adding `false` to the set.